src/mixins/amalgamation-mixin.ts

```diff
@@ -86,49 +94,112 @@ export default class AmalgamationMixin extends Vue {
+    ) {
+      return AmlStatuses.ERROR_FOREIGN_UNLIMITED3
+    }
+    return null
+  }
 
-  // TODO: cannot add BC company if there is a foreign ULC and ted is ULC
+  // NOT CURRENTLY USED
+  // /**
+  //  * Disallow only foreign businesses (including EPs).
+  //  * (An amalgamation where all TINGs are foreign will be Phase 2.)
+  //  */
+  // needBcCompany (): AmlStatuses {
+  //   if (this.isAllForeignOrEp) {
+  //     return AmlStatuses.ERROR_NEED_BC_COMPANY
+  //   }
+  //   return null
+  // }
+
+  /** Disallow if foreign in a short-form horizontal amalgamation. */
+  foreignHorizontal (business: AmalgamatingBusinessIF): AmlStatuses {
+    if (business.type === AmlTypes.FOREIGN && this.isAmalgamationFilingHorizontal) {
+      return AmlStatuses.ERROR_FOREIGN_HORIZONTAL
+    }
+    return null
+  }
 
   //
   // HELPERS
+  // (not all are used atm)
   //
 
+  /** True if all companies in the table are foreign. */
+  get isAllForeign (): boolean {
+    return this.getAmalgamatingBusinesses.every(business => (business.type === AmlTypes.FOREIGN))
+  }
+
+  /** True if all companies in the table are foreign or extra-provincial. */
+  get isAllForeignOrEp (): boolean {
+    return this.getAmalgamatingBusinesses.every(business =>
+      (business.type === AmlTypes.FOREIGN ||
+        (business.type === AmlTypes.LEAR && business.legalType === CorpTypeCd.EXTRA_PRO_A)
+      )
+    )
+  }
+
   /** True if there a foreign company in the table. */
   get isAnyForeign (): boolean {
     return this.getAmalgamatingBusinesses.some(business => (business.type === AmlTypes.FOREIGN))
   }
 
-  /** True if there a CCC in the table. */
+  /** True if there a Benefit Company in the table. */
+  get isAnyBen (): boolean {
+    return this.getAmalgamatingBusinesses.some(business =>
+      (business.type === AmlTypes.LEAR && business.legalType === CorpTypeCd.BENEFIT_COMPANY)
+    )
+  }
+
+  /** True if there is a BC CCC in the table. */
   get isAnyCcc (): boolean {
     return this.getAmalgamatingBusinesses.some(business =>
       (business.type === AmlTypes.LEAR && business.legalType === CorpTypeCd.BC_CCC)
     )
   }
 
-  /** True if there a limited company in the table. */
+  /** True if there is a BC Limited Company in the table. */
   get isAnyLimited (): boolean {
     return this.getAmalgamatingBusinesses.some(business =>
       (business.type === AmlTypes.LEAR && business.legalType === CorpTypeCd.BC_COMPANY)
     )
   }
 
-  /** True if there is an unlimited company in the table. */
+  /** True if there is an BC Unlimited Liability Company in the table. */
   get isAnyUnlimited (): boolean {
     return this.getAmalgamatingBusinesses.some(business =>
       (business.type === AmlTypes.LEAR && business.legalType === CorpTypeCd.BC_ULC_COMPANY)
     )
   }
+
+  /** True if there is a BC company in the table. */
+  get isAnyBcCompany (): boolean {
+    return (this.isAnyBen || this.isAnyCcc || this.isAnyLimited || this.isAnyUnlimited)
+  }
```

Comment:
It is an outstanding question in the requirements document whether a "BC company" means it is type "BC" or whether it means "any bc company". This getter is currently unused but I'm leaving this for now.

Reply:
Sev, what do you think if we move these getters to the store? I'm thinking ahead for when we do short form. But, then again, I guess we can reuse this mixin for short term.

Reply:
Never mind, let's keep as is and see when we get there to short form.

Reply:
Yeah, let's leave it for now. This one may go away anyways.

The other getters in here are specific to the rules in this mixin so I think they're good here.

Reply:
Sounds good.